sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -850,22 +851,29 @@ __ESIMD_API std::
 ///
 template <rgba_channel_mask RGBAMask = rgba_channel_mask::ABGR,
           typename AccessorT, int N,
-          typename T = typename AccessorT::value_type>
+          typename T = typename AccessorT::value_type, typename Toffset>
 __ESIMD_API std::enable_if_t<((N == 8 || N == 16 || N == 32) &&
-                              sizeof(T) == 4 && !std::is_pointer_v<AccessorT>),
+                              sizeof(T) == 4 && !std::is_pointer_v<AccessorT> &&
+                              std::is_integral_v<Toffset>),
                              simd<T, N * get_num_channels_enabled(RGBAMask)>>
-gather_rgba(AccessorT acc, simd<uint32_t, N> offsets,
-            uint32_t global_offset = 0, simd_mask<N> mask = 1) {
+gather_rgba(AccessorT acc, simd<Toffset, N> offsets,
```

Comment:
It seems that vector offsets need similar solution as provided in https://github.com/intel/llvm/pull/9674 , right?

Reply:
Yes, I need to add that fix here, I didn't get to it yet.

Reply:
just added commit with this and other fixes